pterodactyl/client/servers/files.md

```diff
@@ -0,0 +1,259 @@
+
+### `POST /api/client/servers/:serverId/files/write`
+
+Writes the contents of the specified file to the server.
+
+### Body
+
+| Name | Visibility | Type   | Description                       |
+| ---- | ---------- | ------ | --------------------------------- |
+| file | Required   | string | File to write                     |
+| -    | Required   | string | Content of the file to be written |
+
+### Responses
+
+| Code | Description  |
+| ---- | ------------ |
+| 204  | File written |
+
+### `POST /api/client/servers/:serverId/files/compress`
+
+Compresses files on the server.
+
+### Body
+
+| Name    | Visibility | Type          | Description                    |
+| ------- | ---------- | ------------- | ------------------------------ |
+| root    | Required   | string        | Root directory for compression |
+| files[] | Required   | array[string] | Files to compress (multiple)   |
+
+### Responses
+
+| Code | Description                 |
+| ---- | --------------------------- |
+| 200  | Compressed file information |
+
+### `POST /api/client/servers/:serverId/files/decompress`
+
+Decompresses files on the server.
+
+### Body
+
+| Name    | Visibility | Type          | Description                    |
+| ------- | ---------- | ------------- | ------------------------------ |
+| root    | Required   | string        | Root directory for compression |
+| files[] | Required   | array[string] | Files to decompress (multiple) |
+
+### Responses
+
+| Code | Description        |
+| ---- | ------------------ |
+| 204  | Files decompressed |
+
+### `POST /api/client/servers/:serverId/files/delete`
+
+Deletes files or folders on the server.
+
+### Body
+
+| Name    | Visibility | Type          | Description                |
+| ------- | ---------- | ------------- | -------------------------- |
+| root    | Required   | string        | Root directory             |
+| files[] | Required   | array[string] | Files to delete (multiple) |
+
+### Responses
+
+| Code | Description   |
+| ---- | ------------- |
+| 204  | Files deleted |
+
+### `POST /api/client/servers/:serverId/files/create-folder`
+
+Creates a new folder on the server. If the root directory is not provided it defaults to `/`.
+
+### Body
+
+| Name | Visibility | Type   | Description                   |
+| ---- | ---------- | ------ | ----------------------------- |
+| name | Required   | string | Name of the folder            |
+| root | Optional   | string | Root directory (default: '/') |
+
+### Responses
+
+| Code | Description    |
+| ---- | -------------- |
+| 204  | Folder created |
+
+### `POST /api/client/servers/:serverId/files/chmod`
+
+Updates file permissions for file(s) on the server.
+
+### Body
+
+| Name  | Visibility | Type          | Description                 |
+| ----- | ---------- | ------------- | --------------------------- |
+| root  | Required   | string        | Root directory              |
+| files | Required   | array[string] | Files to change permissions |
```

Comment:
This actually takes an array of objects:

```json
[
  {
    "file": "./server.jar",
    "mode": 493
  }
]
```

I'm not sure how we'd go about representing this in the documentation. There might be an example from the Wings documentation that we could use.

Reply:
Alright, I will add another section to show an example of the body.